langchain/src/retrievers/pinecone-hybrid-search.ts

```diff
@@ -0,0 +1,148 @@
+import type {
+  SparseValues,
+  VectorOperationsApi,
+} from "@pinecone-database/pinecone/dist/pinecone-generated-ts-fetch";
+import { BertTokenizer } from "bert-tokenizer";
+
+import { PineconeLibArgs, PineconeMetadata } from "vectorstores/pinecone.js";
+import { Embeddings } from "../embeddings/base.js";
+import { Document } from "../document.js";
+import { BaseRetriever } from "../schema/index.js";
+
+export interface PineconeHybridSearchParams extends PineconeLibArgs {
+  tokenizer: BertTokenizer;
+  topK: number;
+  alpha: number;
+}
+
+export class PineconeHybridSearchRetriever extends BaseRetriever {
+  embeddings: Embeddings;
+
+  pineconeIndex: VectorOperationsApi;
+
+  tokenizer: BertTokenizer;
+
+  topK = 4;
+
+  alpha = 0.5;
+
+  textKey: string;
+
+  namespace?: string;
+
+  filter?: PineconeMetadata;
+
+  constructor(embeddings: Embeddings, args: PineconeHybridSearchParams) {
+    super();
+    this.embeddings = embeddings;
+    this.pineconeIndex = args.pineconeIndex;
+    this.tokenizer = args.tokenizer;
+    this.topK = args.topK;
```

Comment:
_nit_: the default values are being overriden, should that be the case?


```suggestion
    this.topK = args.topK ?? 4;
```